### PR TITLE
ads1x15: refactor, simplify API

### PR DESCRIPTION
**Pull request by maruel**
- Merge PinForDifferenceOfChannels into PinForChannel.
- Make Channel a type.
- Rename ads1x15AnalogPin to analogPin, it was stuttering.
- Shorten few argument names.
- Remove one memory allocation in analogPin.
- Add a minimalistic unit test.

### CODE REVIEW

**Comment by maruel** [comment]
@bezineb5  any strong opinion about `Channel`?

File: experimental/devices/ads1x15/ads1x15.go [hunk 1]
```diff
@@ -20,13 +20,72 @@ import (
...
+	case Channel0:
+		return 0
+	case Channel1:
+		return 1
+	case Channel2:
+		return 2
+	case Channel3:
+		return 3
+	case Channel0Minus1:
+		return 4
```
Comment (maruel) [inline experimental/devices/ads1x15/ads1x15.go]: That's kinda adhoc.

File: experimental/devices/ads1x15/ads1x15.go [hunk 2]
```diff
@@ -383,36 +397,37 @@ var (
 	}
 )
 
-type ads1x15AnalogPin struct {
+type analogPin struct {
 	adc                *Dev
-	query              []byte
+	c                  Channel
+	query              [3]byte
```
Comment (maruel) [inline experimental/devices/ads1x15/ads1x15.go]: This removes one memory allocation.

**Comment by maruel** [comment]
@bezineb5 I would still appreciate your comments on how the API surface changed.

**Comment by bezineb5** [comment]
Hi,

Sorry, I was unexpectedly without internet this week :/.
Your changes are really great! It's much improved now. For the "difference of channel", I agree with the proposal, as the device is anyway too constrained to allow the difference of any pin with any other pin, so the user has to know the device anyway.
I'll do another PR to fix the remaining points (missing a mutex + typo). I'll also add more unit tests to ensure that the decoding is done correctly.

Thanks a lot!

**Comment by maruel** [comment]
Thanks!

**Pull request merged**